daemon/api_notices.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package daemon
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/snapcore/snapd/overlord/auth"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/strutil"
+)
+
+var (
+	noticesCmd = &Command{
+		Path:       "/v2/notices",
+		GET:        getNotices,
+		ReadAccess: openAccess{},
+	}
+
+	noticeCmd = &Command{
+		Path:       "/v2/notices/{id}",
+		GET:        getNotice,
+		ReadAccess: openAccess{},
+	}
+)
+
+func getNotices(c *Command, r *http.Request, user *auth.UserState) Response {
+	query := r.URL.Query()
+
+	typeStrs := strutil.MultiCommaSeparatedList(query["types"])
+	types := make([]state.NoticeType, 0, len(typeStrs))
+	for _, typeStr := range typeStrs {
+		noticeType := state.NoticeType(typeStr)
+		if !noticeType.Valid() {
+			// Ignore invalid notice types (so requests from newer clients
+			// with unknown types succeed).
+			continue
```

Comment:
Good point, thanks! Technically, I'm not sure that this check is even necessary, since the notice filter within `overlord/state/notices.go` will happily accept arbitrary strings, but I agree it's safer to sanity check them here before they make it to the state notices code. I'll add a test for this.